src/_png.cpp

```diff
@@ -131,7 +164,14 @@ static PyObject *Py_write_png(PyObject *self, PyObject *args, PyObject *kwds)
         py_file = filein;
     }
 
-    if ((fp = mpl_PyFile_Dup(py_file, (char *)"wb", &offset))) {
+    if (filein == Py_None) {
+        buff.size = width * height * 4 + 1024;
```

Comment:
Presumably, this is large enough for uncompressed + PNG header and other overhead?